Docs: add Single Sign-On documentation (#455)
SUSE/doc-susemanager: modules/administration/pages/auth-methods.adoc

* Assertion and nameId encryption.
* Assertion signatures.
* Message signatures with AuthNRequest, LogoutRequest, LogoutResponses.
* Enable an Assertion consumer service endpoint.
* Enable a single logout service endpoint.
* Publish the SP metadata (which can be signed).

SSO in {productname} does not support:

* Product choosing and implementation for the Identity Service Provider (IdP).
* SAML support for other products (please check with the respective product documentation).

Before you begin, you will need to have configured an external Identity Service Provider with these parameters.
Check your IdP documentation for instructions.

You will need these endpoints:

* Assertion Consumer Service (or ACS): an endpoint to accept SAML messages to establish a session into the Service Provider.
The endpoint for ACS in {productname} is: https://example.com/rhn/manager/ssaml/acs
* Single Logout Service (or SLS): an endpoint to initiate a logout request from the IdP.
The endpoint for SLS in {productname} is: https://example.com/rhn/manager/ssaml/sls
* Metadata: an endpoint to retrieve {productname} metadata for SAML.

> The endpoint for Metadata in {productname} is https://example.com/rhn/manager/ssaml/metadata

Reviewer: ```suggestion
The endpoint for Metadata in {productname} is: https://example.com/rhn/manager/ssaml/metadata
```
for consistency

[IMPORTANT]
====
Your IdP must have a SAML:Attribute containing the username of the IdP user domain, called `uid`.

> The `uid` attrribute passed in the SAML:Attribute must be created in the {productname} user base before you activate single sign on.

Reviewer: ```suggestion
The `uid` attribute passed in the SAML:Attribute must be created in the {productname} user base before you activate single sign on.
```

====

After the authentication with the IdP using the user `orgadmin` is successful, you will be logged in into {productname} as the `orgadmin` user, provided that the `orgadmin` user exists in {productname}.

Single sign on is disabled by default.

.Procedure: Enabling SSO

. If your users do not yet exist in {productname}, create them first.
. Edit `/etc/rhn/rhn.conf` and add this line at the end of the file:
+
----
java.sson = true
----
. Configure `/usr/share/rhn/config-defaults/rhn_java_sson.conf` with the parameters retrieved from your IdP.

> To find all the occurrences you need to change, search in the file for the placeholders [systemitem]``YOUR-PRODUCT`` and [systemitem]```YOUR-IDP-ENTITY``.

Reviewer: ```suggestion
To find all the occurrences you need to change, search in the file for the placeholders [literal]``YOUR-PRODUCT`` and [literal]```YOUR-IDP-ENTITY``.
```
adoc markup fun

Every parameter comes with a brief explanation of what it is meant for.
. Restart spacewalk-service to pick up the changes:
+
----
spacewalk-service restart
----

> When you visit the {productname} URL, you will be redirected to the IdP for SSON where you will be requested to authenticate.

Reviewer: ```suggestion
When you visit the {productname} URL, you will be redirected to the IdP for SSO where you will be requested to authenticate.
```
